common/src/main/java/com/google/auto/common/Visibility.java

```diff
@@ -76,4 +76,43 @@ public static Visibility effectiveVisibilityOfElement(Element element) {
     }
     return effectiveVisibility;
   }
-}
\ No newline at end of file
+
+  /**
+   * Returns whether the given element is visible from another element in the given package.
+   */
+  public static boolean isVisibleFrom(Element element, PackageElement from) {
+    switch (effectiveVisibilityOfElement(element)) {
+      case PUBLIC:
+        return true;
+      case PROTECTED:
+      case DEFAULT:
+        return MoreElements.getPackage(element).equals(from);
+      case PRIVATE:
+        return false;
+      default:
+        throw new AssertionError();
+    }
+  }
+
+  /**
+   * Returns whether the given element is visible from another element in the given package,
+   * assuming that other element is a subclass (or a member of a subclass) of the former (or
+   * its enclosing type).
```

Comment:
Being a subclass doesn't guarantee `protected` access. If the element being accessed is a constructor or an instance method, it is only accessible from within the same instance. I'm not sure how to capture that here. Maybe just give in and actually mention `protected`.